### PR TITLE
Remove all io/ioutil imports.

### PR DESCRIPTION
**Pull request by cmacknz**
What the title says. io/ioutil is deprecated and I kept seeing the linter fail because of it, so remove all imports of io/ioutil. The linter will warn us if we bring it back.

For the most part this was just simple renames, except the conversion to os.ReadDir which changes the return type.

### CODE REVIEW

File: dev-tools/mage/copy.go [hunk 1]
```diff
@@ -104,20 +104,20 @@ func (t *CopyTask) fileCopy(src, dest string, info os.FileInfo) error {
...
 	if mode == 0 {
-		mode = info.Mode()
+		mode = entry.Type()
 	}
 	if err := os.MkdirAll(dest, mode&os.ModePerm); err != nil {
 		return errors.Wrap(err, "failed creating dirs")
 	}
 
-	contents, err := ioutil.ReadDir(src)
+	contents, err := os.ReadDir(src)
```
Comment (cmacknz) [inline dev-tools/mage/copy.go:120]: Conversion to os.ReadDir, the return type changed to fs.DirEntry. Double check this code.

File: dev-tools/mage/gotest.go [hunk 2]
```diff
@@ -126,14 +125,14 @@ func DefaultTestBinaryArgs() TestBinaryArgs {
 // Use MODULE=module to run only tests for `module`.
 func GoTestIntegrationForModule(ctx context.Context) error {
 	module := EnvOr("MODULE", "")
-	modulesFileInfo, err := ioutil.ReadDir("./module")
+	modulesDirEntry, err := os.ReadDir("./module")
```
Comment (cmacknz) [inline dev-tools/mage/gotest.go:128]: Conversion to os.ReadDir, the return type changed to fs.DirEntry. Double check this code.

**Comment by cmacknz** [comment]
There was only one place that resulted in more code changes than just a simple rename, I've commented it on it to point out where more careful review is required.

**Review by blakerouse — APPROVED** [review]
Thanks for this change.

I reviewed the mage copy code, nothing stood out to me that maybe me think the logic didn't match and work the same.

**Comment by cmacknz** [comment]
Yeah that Copy change broke mage package but I don't see how yet.

**Review by michalpristas — APPROVED** [review]

**Comment by cmacknz** [comment]
Only failure is the Windows 10 unit tests failing to get a buildkite agent to run on. https://buildkite.com/elastic/elastic-agent/builds/6439

The windows 11 unit tests and integration tests passed, force merging.

**Pull request merged**